fix(deps): pin fff-search git deps to known-good rev

### Description

`Cargo.lock` is gitignored, and the three `fff-*` git dependencies in the root `Cargo.toml` (`fff-search`, `fff-grep`, `fff-query-parser`) are declared without a pinned `rev`. Every CI run re-resolves them to the upstream HEAD. The upstream repo just changed `grep_search`'s visibility and signature, breaking `crates/app/src/tools/fff_grep.rs`:

\`\`\`
error[E0603]: function \`grep_search\` is private
  --> crates/app/src/tools/fff_grep.rs:140:24
\`\`\`

This blocks CI on every open PR. It is not caused by any of the open PRs; it surfaces on whichever one runs next.

### Minimal fix

Pin all three deps to the last-known-good SHA by adding `rev = "ea1f9802d7879345f02bef72bcadecd7252b8add"` to each declaration in the workspace root `Cargo.toml`. That SHA is what current developers' local `Cargo.lock` files resolve to. Picking a pin is independent of the debate over whether to unignore `Cargo.lock` workspace-wide.

### Component

core (workspace dep resolution)

### Alternatives considered

- Unignore `Cargo.lock` — correct per Rust convention for binary crates, but much larger change (affects every future PR); do in a separate PR after this pin lands.
- Update `crates/app/src/tools/fff_grep.rs` to the new upstream API — would unblock CI but does not prevent the next upstream break; still need the pin.

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -275,10 +275,17 @@ rara-skills = { path = "crates/skills" }
 rara-soul = { path = "crates/soul" }
 rara-vault = { path = "crates/rara-vault" }
 
-# fff (Fast File Finder) — Cargo git dependencies (not vendored)
-fff-search = { git = "https://github.com/dmtrKovalenko/fff.nvim", features = ["zlob"] }
-fff-grep = { git = "https://github.com/dmtrKovalenko/fff.nvim" }
-fff-query-parser = { git = "https://github.com/dmtrKovalenko/fff.nvim", default-features = false }
+# fff (Fast File Finder) — Cargo git dependencies (not vendored).
+#
+# Pinned to a specific rev because `Cargo.lock` is gitignored workspace-wide
+# (see `.gitignore`); without the pin CI re-resolves to upstream HEAD on
+# every run and a breaking change upstream (e.g. the `grep_search` visibility
+# flip in Apr 2026) takes down every open PR at once. Bump the rev in a
+# dedicated PR after updating `crates/app/src/tools/{fff_find,fff_grep}.rs`
+# to the new upstream API.
+fff-search = { git = "https://github.com/dmtrKovalenko/fff.nvim", rev = "ea1f9802d7879345f02bef72bcadecd7252b8add", features = ["zlob"] }
+fff-grep = { git = "https://github.com/dmtrKovalenko/fff.nvim", rev = "ea1f9802d7879345f02bef72bcadecd7252b8add" }
+fff-query-parser = { git = "https://github.com/dmtrKovalenko/fff.nvim", rev = "ea1f9802d7879345f02bef72bcadecd7252b8add", default-features = false }
 
 rara-tool-macro = { path = "crates/common/tool-macro" }
 yunara-store = { path = "crates/common/yunara-store" }

